x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/dataframe/DataFrameAnalyticsState.java

```diff
@@ -29,11 +28,7 @@ public static DataFrameAnalyticsState fromStream(StreamInput in) throws IOExcept
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         DataFrameAnalyticsState toWrite = this;
-        if (out.getVersion().before(Version.V_7_5_0) && toWrite == STARTING) {
-            // Before 7.5.0 there was no STARTING state and jobs for which
-            // tasks existed but were unassigned were considered STOPPED
-            toWrite = STOPPED;
-        }
+
         out.writeEnum(toWrite);
```

Comment:
There's no need to have a separate local variable now.

```suggestion
        out.writeEnum(this);
```

plus delete the line `DataFrameAnalyticsState toWrite = this;`